### DISCUSSION TITLE
pybamm sol.plot() incompatible with matplotlib.pyplot

### DISCUSSION THREAD

**Discussion by jennya2a** [pybamm-team/PyBaMM#3841] [Q&A] [answered]
I run a DFN model, and my 1st figure is using sol.plot(), my 2nd figure is using pyplot. 
But my 2nd figure only show up when I closed the 1st figure.
Could any one tell me how to fix it to show all figures without closing anyone.
Thanks.

**Comment by valentinsulzer** [comment, marked as answer]
Do `sol.plot(testing=True)` (we should rename that argument)

**Reply by jennya2a** [reply]
it works! Thanks. could you explain what does 'testing=True' mean here?

**Reply by agriyakhetarpal** [reply]
The `testing=True` argument gets supplied further to the `dynamic_plot` method in  the `pybamm.QuickPlot` module, and is documented here: https://docs.pybamm.org/en/stable/source/api/plotting/quick_plot.html#pybamm.QuickPlot.dynamic_plot

**Reply by valentinsulzer** [reply]
It's just a boolean input that decides whether we call `plt.show()` within the plotting function or not. We called it "testing" because the original use case for using this argument is to avoid calling `plt.show()` when running unit tests (default is testing=False: call `plt.show()`, when testing=True, don't call it). This will be renamed to `show_plot` in the next version (#3842 ) to clarify what it does (call `plt.show()` only if `show_plot=True`)
